src/cmd/singularity/actions_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"testing"
+
+	"github.com/sylabs/singularity/src/pkg/test"
+)
+
+//build base image for tests
+const imagePath = "./container.img"
+
+type execOpts struct {
+	binds   []string
+	contain bool
+	home    string
+	workdir string
+	pwd     string
+}
+
+func imageExec(opts execOpts, imagePath string, command []string) ([]byte, error) {
```

Comment:
What is the purpose of this function if it never gets called?

Reply:
that func is used by `build_test.go` and `docker_test.go` , not in this file. as we need to catch the exit code